elessar/pages/Login/login.tsx

```diff
@@ -1,51 +1,56 @@
-import React, { Component, useState, useEffect, Props }  from 'react';
-import { View, Keyboard, Image, StyleSheet, TouchableOpacity, Text, TextInput} from 'react-native';
-import './validation.tsx'
+import React, { Component} from 'react';
+import { View, Keyboard, ActivityIndicator, StyleSheet, TouchableOpacity, Text, TextInput } from 'react-native';
 import { validateEmail, validatePassword } from './validation';
-import { storeData, retrieveData } from './store';
-
-import { AsyncStorage } from 'react-native';
-import gql from 'graphql-tag';
-import { client } from '../../App';
-import { getToken } from './requests';
-
-
-
+import { storeData } from './store';
+import { getToken } from './requests'
 
 interface LoginState {
     email: string,
     message: string,
-    password:string,
+    password: string,
+    loading: boolean,
+}
+
+interface Props {
+    navigation: any
   }
 
-export default class Login extends Component<{},LoginState>{
+export default class Login extends Component<Props, LoginState>{
     constructor(props) {
         super(props);
         this.state = {
             email: '',
             message: '',
             password: '',
+            loading: false
         }
     }
 
-    handleButton = async () => {
+    private handleButton = async () => {
+
 
         const isEmailValid = validateEmail(this.state.email);
         const isPasswordValid = validatePassword(this.state.password);
         
         const emailMessage = isEmailValid ? '' : 'E-mail inválido';
         let passwordMessage = isPasswordValid ? '' : 'Senha inválida, deve conter pelo menos 7 caracteres e conter 1 letra e 1 número.';
-        passwordMessage = !isEmailValid && !isPasswordValid ? '\n' + passwordMessage;
+        passwordMessage = !isEmailValid && !isPasswordValid ? '\n' + passwordMessage: passwordMessage;
         
         this.setState({ message: emailMessage + passwordMessage });
 
         if(isEmailValid && isPasswordValid){
             const requestResult = await getToken(this.state.email, this.state.password);
-            if(!requestResult.data?.login?.token){
+            if(typeof requestResult.data?.login?.token == 'undefined'){
```

Comment:
Por que a mudança?

Reply:
Ele não estava respeitando a condição não lembro exatamente o motivo, mas fiquei em dúvida como o JS trata variáveis não booleanas como booleanos. O JS trata qlqr coisa diferente de 0 ou false como true?
Por exemplo isso tudo em pseudocódigo ficaria:
a) if (not(result.data existe) && not(login existe) && not(token existe))
ou
b) if not((result.data existe) && (login existe) && (token existe))
Qual das duas? Porque a segunda o not transforma o && em ||.
Achei a condição confusa demais e decidi simplificar, o que inclui se o result é vazio por conta de problemas de Internet.

Reply:
+ [Esse](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) link pode te ajudar com isso
+ E [esse](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness) aqui com comparações